scheduler/scheduler-server/src/main/java/org/ow2/proactive/scheduler/core/SchedulingService.java

```diff
@@ -779,14 +779,16 @@ void terminateJobHandling(final JobId jobId) {
     }
 
     private void recover(RecoveredSchedulerState recoveredState) {
+        Vector<InternalJob> finishedJobs = recoveredState.getFinishedJobs();
         Vector<InternalJob> pendingJobs = recoveredState.getPendingJobs();
         Vector<InternalJob> runningJobs = recoveredState.getRunningJobs();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Replace the synchronized class "Vector" by an unsynchronized one such as "ArrayList" or "LinkedList". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://nemo.sonarqube.org/coding_rules#rule_key=squid%3AS1149)